docs/upgrade.rst

```diff
@@ -37,9 +36,13 @@ If ``__eq__`` defined but not ``__hash__``, ``__hash__`` is now set to
 ``None``, as in normal CPython. You should add ``__hash__`` if you intended the
 class to be hashable, possibly using the new ``py::hash`` shortcut.
 
-Usage of the ``PYBIND11_OVERLOAD*`` macros and ``get_overload`` function should
-be replaced by ``PYBIND11_OVERRIDE*`` and ``get_override``. In the future, the
-old macros may be deprecated and removed.
+The constructors for ``py::array`` now always take signed integers for size,
+for consistency. Can cause some compiler warnings to appear. Cast to
```

Comment:
This may lead to compiler warning on some systems.
?

Reply:
Tried to reword a bit. It's a bit tricky to say properly. For me, this is the biggest thing I notice when upgrading 2.5 -> 2.6, this starts spewing warnings. It's about 50/50 on how many `size_t` casts I have to remove and how many `py::ssize_t` casts I have to add, I think.